versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5846 @@
+}
+```
+
+Also note that it is impossible to have a multi-level optional type, e.g. `Int??`; thus, the outputs of a conditional block are only ever single-level optionals, even when there are nested conditionals.
+
+```wdl
+workflow foo {
+  input {
+    Boolean b
+    Boolean c
+  }
+
+  if(b) {
+    if(c) {
+      call x
+      Int x_out = x.out
+    }
+  }
+
+  # Even though it's within a nested conditional, x_out
+  # has a type of Int? rather than Int??
+  Int? x_out_maybe = x_out 
+
+  # Call 'y' which takes an Int input.
+  # The select_first produces an Int, not an Int?
+  call y { input: int_input = select_first([x_out_maybe, 5]) } 
+}
+```
+
+Remember that optional types can be coalesced by using the `select_all` and `select_first` functions. For example:
+
+```wdl
+workflow foo {
+  input {
+    Array[Int] scatter_range = [1, 2, 3, 4, 5]
+  }
+
+  scatter (i in scatter_range) {
+    call x { input: i = i }
+    if (x.validOutput) {
+      Int x_out = x.out
+    }
+  }
+
+  # Because it was declared inside the scatter and the if-block, 
+  # the type of x_out is different here:
+  Array[Int?] x_out_maybes = x_out
+
+  # We can select only the valid elements with select_all:
+  Array[Int] x_out_valids = select_all(x_out_maybes)
+
+  # Or we can select the first valid element:
+  Int x_out_first = select_first(x_out_maybes)
+}
+```
+
+## Struct Definition
+
+A struct type is a user-defined data type. Structs enable the creation of compound data types that bundle together related attributes in a more natural way than is possible using the general-purpose compound types like `Pair` or `Map`. Once defined, a struct type can be used as the type of a declaration like any other data type.
+
+A struct is a top-level WDL element, meaning it is defined at the same level as tasks and workflows, and it cannot be defined within a task or workflow body. A struct is defined using the `struct` keyword, followed by a name that is unique within the WDL document, and a body containing a set of member declarations. Declarations in a `struct` body differ from those in a task or workflow in that struct members cannot have default initializers. Struct members may be optional.
+
+Valid struct:
+```wdl
+struct Person {
+  String first_name
+  String last_name
+  Int age
+  Float? income
+}
+```
+
+Invalid struct:
+```wdl
+struct Invalid {
+  String myString = "Cannot do this"
+  Int myInt
+}
+```
+
+A struct member may be of any type, including compound types and even other struct types.
+
+```wdl
+struct Sample {
+  String id
+  Person donor
+  Map[String, Array[File]] assay_data
+}
+```
+
+### Struct Literals
+
+A struct literal is similar to an [`object` literal](#-object), except that the struct name is used in place of the `object` keyword. The members of a struct value can be validated against the struct's definition at the time of creation, which makes WDL code written using `struct`s more amenable to static type analysis.
```

Comment:
Beware referring to to-be-deprecated text without mentioning it's status.

Reply:
fixed